### PR TITLE
Add generate debug dump function

### PR DESCRIPTION
**Pull request by itaibaz**
Function creates debug dump file, for debugging purposes
This file collects SAI state information and vendor SDK information

### CODE REVIEW

**Comment by lguohan** [comment]
test

**Comment by lguohan** [comment]
test

**Comment by lguohan** [comment]
plan to check-in by the end of this week.

File: inc/sai.h [hunk 1]
```diff
@@ -207,6 +207,16 @@ sai_object_type_t sai_object_type_query(
         _In_ sai_object_id_t sai_object_id);
 
 /**
+* @brief Generate dump file. The dump file may include SAI state information and vendor SDK information.
+*
+* @param[in] dump_file_name Full path for dump file
+*
+* @return #SAI_STATUS_SUCCESS on success Failure status code on error
+*/
+sai_status_t sai_dbg_generate_dump(
```
Comment (kcudnik) [inline inc/sai.h]: plus spaces before "*" to allign all * in column

**Pull request merged**